Feat/kivoyo eks selective deployment

benie-joy-possi opened the pull request:
description here: #26

benie-joy-possi committed:
feat(ami): Add K-Server custom AMI and refine Karpenter AMI module

benie-joy-possi committed:
feat(eks): Add K-Server node group and CloudNativePG operator integration

benie-joy-possi committed:
feat(custom-ami): Introduce custom script support for AMI builder module

benie-joy-possi committed:
refactor(tf-aigateway): Remove redundant Envoy Gateway config module

benie-joy-possi committed:
feat(outputs,variables): Add K-Server AMI ID output and related variables

benie-joy-possi committed:
feat(scripts): Add apply_kivoyo_eks.sh for targeted Terraform application

benie-joy-possi committed:
Refactor apply_kivoyo_eks.sh for improved robustness and clarity. This includes adding pre-requisite command checks, streamlining Terraform backend initialization, dynamically incorporating .tfvars files, and implementing post-apply Kubernetes resource verification.

Koufan-De-King commented on files/k_server_customization_script.sh:
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+echo "Running k_server custom AMI customization script"
+# Add any necessary commands here to customize the AMI for k_server
+# For example, installing specific packages or configuring settings.
```
Is this file necessary?

benie-joy-possi replied:
I added this file just in case we will need to customise the AMI. And i made it optional!

marcjazz commented on apply_kivoyo_eks.sh:
Could you offer options to run this script in a non interactive way?

benie-joy-possi replied:
From my experience a terraform.tfstate.backup file in your working directory is created when you run a terraform plan or terraform apply and there changes to be made which will likely update the current state of the infrastructure.

marcjazz reviewed (changes requested):
Group your script files under same directory.

Franck-Sorel reviewed (commented):
Looks Okay !

benie-joy-possi commented:
@Franck-Sorel  I don't quite get your question. Kindly ellaborate please!

benie-joy-possi committed:
feat: add auto-approve flag to apply_kivoyo_eks.sh

- Add --auto-approve for non-interactive execution

benie-joy-possi committed:
refactor: reorganize scripts into dedicated directory

- Moved all shell scripts to a new  directory
- Updated path to k_server_customization_script.sh in ami_builder.tf
- Added comprehensive README.md in scripts directory

marcjazz commented on apply_kivoyo_eks.sh:
```diff
@@ -0,0 +1,207 @@
+    "module.kserve"               # Deploys KServe components onto the EKS cluster.
+    "module.ai_gateway"           # Deploys Envoy/AI Gateway components.
+    "module.lmcache"              # Deploys LMCache components.
+)
```
I should be able to override theses modules when calling the script. You are at least not planning to write a single script for each selective apply!?

marcjazz commented on scripts/apply_kivoyo_eks.sh:
```diff
@@ -0,0 +1,207 @@
+    "module.kserve"               # Deploys KServe components onto the EKS cluster.
+    "module.ai_gateway"           # Deploys Envoy/AI Gateway components.
+    "module.lmcache"              # Deploys LMCache components.
+)
```
I should be able to override this when calling the script. You are at least not planning to write a different script for each selective apply.

benie-joy-possi replied:
The script has been updated now  @marcjazz  in regards to that.

stephane-segning reviewed (commented):
Wow! Did you tested it already on any aws account?

benie-joy-possi committed:
feat: enhance terraform apply script with flexible module targeting and verify-only mode

benie-joy-possi committed:
feat(): update the readme file accorsing to changes made in the selective deployment script

Franck-Sorel commented on scripts/apply_kivoyo_eks.sh:
It would be beneficial to add a check at the beginning of the script to ensure that at least one module is selected for application. This would prevent the script from running unnecessarily if no modules are specified.

Franck-Sorel commented:
I've reviewed this PR, and it looks good overall. It aligns with the goal of issue #26 to enable selective deployment of EKS components. The new `apply_kivoyo_eks.sh` script is a great addition for managing deployments.

I have a few suggestions for improvement:

1.  The `k_server_customization_script.sh` is currently a placeholder. It would be helpful to add a comment explaining its intended purpose or what kind of customizations are expected.
2.  The `envoy_gateway_config` module has been removed. Could you please provide some context on why this is no longer needed?
3.  The default value for `enable_ai_gateway_redis` has been changed to `false`. It would be great to understand the reasoning behind this change.

I'll also add some specific comments to the code. Let me know what you think!

Franck-Sorel reviewed (commented):
Great work on this PR! I've left a few comments with some suggestions for improvement.

Koufan-De-King committed:
Merge branch 'main' into feat/kivoyo-eks-selective-deployment

benie-joy-possi committed:
feat: created a terraform configuration that creates an eks cluster for testing purposes

benie-joy-possi committed:
Merge branch 'feat/kivoyo-eks-selective-deployment' of https://github.com/ADORSYS-GIS/ai-ops-terraform into feat/kivoyo-eks-selective-deployment

Koufan-De-King committed:
removed

Koufan-De-King committed:
removed 2

Koufan-De-King committed:
chore(front-end): changed logic to use backend business date if available

Koufan-De-King committed:
chore(front-end): changed logic

Koufan-De-King committed:
chore(): added time between image builing and image id fetching

Koufan-De-King committed:
feat(tf): changes to tf structure

stephane-segning committed:
structure update

stephane-segning committed:
chore: structure

stephane-segning committed:
chore: tf test

Koufan-De-King committed:
before modification

Koufan-De-King committed:
before modification 2

Koufan-De-King committed:
more modifications

stephane-segning committed:
fix: aws sg + iam permission fix

Koufan-De-King committed:
more modifications

Koufan-De-King committed:
more modifications

Koufan-De-King committed:
used aws cli to change build instance storage capacity

Koufan-De-King committed:
used a custom launch template for custom ami instances

stephane-segning committed:
Merge pull request #30 from ADORSYS-GIS/feat/kivoyo-eks-selective-deployment--v2

Feat/kivoyo eks selective deployment  v2

stephane-segning committed:
feat: Add model packages module with Helm chart and Karpenter integration

- Introduced a new module for managing model packages, including EFS, vLLM deployments, and Karpenter node pools.
- Created Helm chart structure with templates for PersistentVolumes, PersistentVolumeClaims, Deployments, and Services.
- Implemented configuration for Karpenter node classes and pools based on model specifications.
- Removed legacy VPC and EKS module configurations from mods.tf.
- Updated secret management to reference new namespace structure.
- Added variables for model package configuration and enabled flag.
- Deleted unused namespace resources to streamline deployment.

stephane-segning committed:
refactor: Update Kubernetes resource definitions to use kubernetes_namespace_v1 and remove deprecated files

stephane-segning committed:
feat: Enhance Terraform configurations for model packages and EKS deployment

stephane-segning committed:
feat: Add EC2 Image Builder module with AMI build and wait functionality

stephane-segning committed:
feat: Implement EBS and EFS storage options for model packages and enhance customization script

stephane-segning committed:
feat: Update AMI configuration and customization script for improved EKS deployment

stephane-segning committed:
feat: Enhance AMI build process with VPC integration and add SSM endpoints

stephane-segning committed:
feat: Refactor locals and remove unused Kubernetes namespaces; enhance Image Builder configuration

stephane-segning committed:
feat: Update image tags and enhance custom AMI configuration for improved deployment

stephane-segning committed:
feat: Enhance documentation and developer workflow; update custom AMI configuration for improved deployment

stephane-segning committed:
feat: Add README documentation for ami-build, eks, model-packages, and vpc modules; enhance clarity on functionality and usage

stephane-segning committed:
feat: Enhance module documentation for ami-build, eks, model-packages, and vpc; add requirements, providers, inputs, and outputs sections

stephane-segning committed:
feat: Add support for exporting model-packages Helm values to S3 for Argo/GitOps; include new variables and resources
